pysmt/smtlib/script.py

```diff
@@ -290,6 +333,43 @@ def __str__(self):
         return "\n".join((str(cmd) for cmd in self.commands))
 
 
+def _parse_goal(command, max_smt_goals=None, goal_position=None):
+    assert len(command.args) == 2, f"Command {command.name} must have 2 arguments"
+    singed = False
+    if len(command.args) >= 2:
+        options = command.args[1]
+        if options is not None:
+            for arg in options:
+                if arg[0] == ":signed":
+                    singed = arg[1]
+                break
+    if command.name == smtcmd.MAXIMIZE:
+        return MaximizationGoal(command.args[0], singed)
+    elif command.name == smtcmd.MINIMIZE:
+        return MinimizationGoal(command.args[0], singed)
+    elif command.name == smtcmd.MINMAX:
+        return MinMaxGoal(command.args[0], singed)
+    elif command.name == smtcmd.MAXMIN:
+        return MaxMinGoal(command.args[0], singed)
+    elif command.name == smtcmd.ASSERT_SOFT:
+        assert max_smt_goals is not None
+        assert goal_position is not None
+        formula = command.args[0]
+        cmd_annotations = dict(command.args[1])
+        max_smt_id = cmd_annotations.get(":id", "")
+        max_smt_weight = cmd_annotations.get(":weight", 1)
+        if not isinstance(max_smt_weight, int):
+            max_smt_weight = max_smt_weight.constant_value()
+        _, goal = max_smt_goals.setdefault(max_smt_id, (goal_position, MaxSMTGoal()))
+        goal.add_soft_clause(formula, max_smt_weight)
+        if len(goal.soft) == 1:
+            return goal
+        # Return None because the max_smt_goal is already in the goal list
```

Comment:
This is quite a strange use of `None`. consider inlining this function, since you separately consider all the cases anyway...